Mathlib/Algebra/Module/Equiv.lean

```diff
@@ -122,6 +122,19 @@ instance (priority := 100) [RingHomInvPair σ σ'] [RingHomInvPair σ' σ]
   [EquivLike F M M₂] [s : SemilinearEquivClass F σ M M₂] : SemilinearMapClass F σ M M₂ :=
   { s with }
 
+variable {F}
+
+/-- Reinterpret an element of a type of semilinear equivalences as a semilinear equivalence. -/
+@[coe]
+def semilinearEquiv [RingHomInvPair σ σ'] [RingHomInvPair σ' σ]
+    [EquivLike F M M₂] [SemilinearEquivClass F σ M M₂] (f : F) : M ≃ₛₗ[σ] M₂ :=
+  { (f : M ≃+ M₂), (f : M →ₛₗ[σ] M₂) with }
+
+/-- Reinterpret an element of a type of semilinear equivalences as a semilinear equivalence. -/
+instance instCoeToSemilinearEquiv [RingHomInvPair σ σ'] [RingHomInvPair σ' σ]
+    [EquivLike F M M₂] [SemilinearEquivClass F σ M M₂] : CoeHead F (M ≃ₛₗ[σ] M₂) where
```

Comment:
```suggestion
    [EquivLike F M M₂] [SemilinearEquivClass F σ M M₂] : CoeTC F (M ≃ₛₗ[σ] M₂) where
```
For `Equiv` and `MulEquiv` this was done using `CoeTC`, so I imagine we want that everywhere?

bors d+

Reply:
I don't really understand this stuff but judging by [this comment](https://github.com/leanprover-community/mathlib4/pull/10368#discussion_r1488584069) it looks like `CoeHead` is preferred here